java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -89,130 +152,99 @@ public String getBuild() {
         }
     }
 
-    protected String getEnvironmentVariable(String key) {
-        return System.getenv(key);
-    }
-
-    public SauceOptions withChrome() {
-        browserName = BrowserType.CHROME;
-        return this;
-    }
-
-    public SauceOptions withSafari() {
-        return withMac(MacVersion.Mojave);
-    }
-
-    public SauceOptions withSafari(SafariVersion version) {
-        browserName = BrowserType.SAFARI;
-        browserVersion = version.getVersion();
-        return this;
-    }
-
-    public SauceOptions withSafari(final String version) {
-        String _version = version;
-        if (_version.isEmpty()) {
-            _version = "latest";
+    // Use Case is pulling serialized information from JSON/YAML, converting it to a HashMap and passing it in
+    // This is a preferred pattern as it avoids conditionals in code
+    public void setCapabilities(Map<String, Object> capabilities) {
+        capabilities.forEach(this::setCapability);
+    }
+
+    // This might be made public in future version; For now, no good reason to prefer it over direct accessor
+    private Object getCapability(String capability) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        String getter = "get" + capability.substring(0, 1).toUpperCase() + capability.substring(1);
+        Method declaredMethod = SauceOptions.class.getDeclaredMethod(getter);
+        return declaredMethod.invoke(this);
+    }
+
+    // This might be made public in future version; For now, no good reason to prefer it over direct accessor
+    public void setCapability(String key, Object value) {
+        if (Options.primaryEnum.contains(key) && value.getClass().equals(String.class)) {
+            setEnumCapability(key, (String) value);
+        } else if ("prerun".equals(key) && ((HashMap) value).keySet().toArray()[0].getClass().equals(String.class)) {
+            Map<Options.Prerun, Object> prerunMap = new HashMap<>();
```

Comment:
Amusingly I just pulled them back into this method because I wasn't sure why they were specified outside. There's no additional efficiency to it, but easy enough to do.

Reply:
done.